packages/apputils-extension/src/notificationplugin.tsx

```diff
@@ -0,0 +1,856 @@
+    super();
+    this._count = manager.count;
+    this.manager.changed.connect(this.onNotificationChanged, this);
+  }
+
+  /**
+   * Number of notifications.
+   */
+  get count(): number {
+    return this._count;
+  }
+
+  /**
+   * Whether to silence all notifications or not.
+   */
+  get doNotDisturbMode(): boolean {
+    return this._doNotDisturbMode;
+  }
+  set doNotDisturbMode(v: boolean) {
+    this._doNotDisturbMode = v;
+  }
+
+  /**
+   * Whether to highlight the status widget or not.
+   */
+  get highlight(): boolean {
+    return this._highlight;
+  }
+
+  /**
+   * Whether the popup is opened or not.
+   */
+  get listOpened(): boolean {
+    return this._listOpened;
+  }
+  set listOpened(v: boolean) {
+    this._listOpened = v;
+    if (this._listOpened || this._highlight) {
+      this._highlight = false;
+    }
+    this.stateChanged.emit();
+  }
+
+  protected onNotificationChanged(
+    manager: NotificationManager,
+    change: Notification.IChange
+  ): void {
+    // Set private attribute to trigger only once the signal emission
+    this._count = this.manager.count;
+
+    const { autoClose } = change.notification.options;
+    const noToast =
+      this.doNotDisturbMode ||
+      (typeof autoClose === 'number' && autoClose <= 0);
+
+    // Highlight if
+    //   the list is not opened (the style change if list is opened due to clickedItem style in statusbar.)
+    //   the change type is not removed
+    //   the notification will be hidden
+    if (!this._listOpened && change.type !== 'removed' && noToast) {
+      this._highlight = true;
+    }
+    this.stateChanged.emit();
+  }
+
+  private _count: number;
+  private _highlight = false;
+  private _listOpened = false;
+  private _doNotDisturbMode = false;
+}
+
+/**
+ * Status view properties
+ */
+interface INotificationStatusProps {
+  /**
+   * Number of notification
+   */
+  count: number;
+  /**
+   * Whether to highlight the view or not.
+   */
+  highlight: boolean;
+  /**
+   * Click event handler
+   */
+  onClick: () => void;
+  /**
+   * Translation object
+   */
+  trans: TranslationBundle;
+}
+
+/**
+ * Status view
+ */
+function NotificationStatus(props: INotificationStatusProps): JSX.Element {
+  return (
+    <GroupItem
+      spacing={HALF_SPACING}
+      onClick={() => {
+        props.onClick();
+      }}
+      title={
+        props.count > 0
+          ? props.trans._n('%1 notification', '%1 notifications', props.count)
+          : props.trans.__('No notifications')
+      }
+    >
+      <TextItem
+        className="jp-Notification-Status-Text"
+        source={`${props.count}`}
+      ></TextItem>
+      <bellIcon.react top={'2px'} stylesheet={'statusBar'}></bellIcon.react>
+    </GroupItem>
+  );
+}
+
+/**
+ * Add notification center and toast
+ */
+export const notificationPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/apputils-extension:notification',
+  autoStart: true,
+  requires: [IStatusBar],
```

Comment:
Is `IStatusBar` really required for this plugin? Or could it also be made optional?

Or move the status bar related logic to a separate plugin so the commands are still available.

Reply:
The only UI entry point for the notification is a status bar widget - as for the log console.

I don't know what would be the best solution for Notebook v7 - as we should promote not disturbing the user with many toasts.

This definitely needs to be discussed for notebook v7 (including the log console that I guess is not loaded neither for now).

Reply:
Although it would be great to have to leverage announcements features in nb7 like https://github.com/jupyterlab/jupyterlab/pull/13365